Tte (#85)
GSK-Biostatistics/beastt: vignettes/tte.Rmd

---

> title: "Weibull Endpoint with Borrowing from an External Control Arm"

Reviewer: ```suggestion
title: "Time-to-Event Endpoint with Borrowing from an External Control Arm"
```

```{r}
prop_scr_love(ps_obj, reference_line = 0.1)
```

```{r}
pwr_prior <- calc_power_prior_weibull(
  ps_obj,
  y, event, 
  dist_normal(0, 10),
  50,
  approximation = "MCMC"
)
plot_dist(pwr_prior)
```

Now that we have a normal power prior, we can calculate the posterior distribution for $\theta_C$ using the `calc_post_norm` function. By defining our prior to be a normal distribution and by assuming the SD of the internal response data to be known, the resulting posterior distribution will also be normal (the case when the SD is unknown is not covered in this vignette).

**Note: if reading internal data directly into `calc_post_norm` instead of a propensity score object, be sure to include only the treatment arm of interest (e.g., the internal control arm if creating a posterior distribution for $\theta_C$).**

```{r}
post <- calc_post_norm(ps_obj,
                       response = y, 
                       prior = pwr_prior,
                       internal_sd = sd_internal_control)

> plot_dist(post)

Reviewer: These lines probably won't run, so we can delete them for now. I'll update this vignette in an upcoming pull request.

```

If we want to robustify our power prior for $\theta_C$, we can add a vague component to the power prior distribution we previously created to construct a robust mixture prior which we can then pass to `calc_post_norm`. In general, we can define our prior to be a mixture distribution with an arbitrary number of normal and/or $t$ components. If any component of the prior is a $t$ distribution, the prior will be approximated with the mixture of two normal distributions.

```{r}
mixed_prior <- robustify_norm(pwr_prior, n_external)
plot_dist("Robust Mixture Prior" = mixed_prior,
          "Power Prior" = pwr_prior)
```

```{r}

> cont_post <- calc_weibull_post(filter(int_tte_df, trt == 0),

Reviewer: ```suggestion
cont_post <- calc_post_weibull(filter(int_tte_df, trt == 0),
```

                  y, event, rpp, c(10, 12))

vague_only <- dist_mixture(parameters(rpp)$dist, weights = c(0,1))

> trt_post <- calc_weibull_post(filter(int_tte_df, trt == 1),

Reviewer: ```suggestion
trt_post <- calc_post_weibull(filter(int_tte_df, trt == 1),
```
